python/ray/_private/utils.py

```diff
@@ -465,8 +470,23 @@ def _get_docker_cpus(
         except Exception as e:
             logger.exception("Unexpected error calculating docker cpu quota.",
                              e)
+    # Look at cpu.max for cgroups v2
+    elif os.path.exists(cpu_max_file_name):
+        try:
+            max_file = open(cpu_max_file_name).read()
+            quota_str, period_str = max_file.split()
+            if quota_str.isnumeric() and period_str.isnumeric():
+                cpu_quota = float(quota_str) / float(period_str)
+            else:
+                # quota_str is "max" meaning the cpu quota is unset
+                cpu_quota = None
+        except Exception:
+            logger.exception("Unexpected error calculating docker cpu quota.")
     if (cpu_quota is not None) and (cpu_quota < 0):
```

Comment:
I think we can rephrase these in a single if statement? (also i don't think we're allowed to have fractional cpu's in the total at all?) 

```
if cpu_quota is not None and cpu_quota >= 0: 
  cpu_quota = math.ceil(cpu_quota, 1)
```

Reply:
That would miss the functionality of setting a negative cpu_quota to None.
Btw, is it actually possible to have negative cpu quota or is this defensive programming?

Reply:
i don't think so? or maybe there's a possibility of `-1` floating around out there? maybe ignore this comment actually, may have been an over-eager attempt to simplify this logic.